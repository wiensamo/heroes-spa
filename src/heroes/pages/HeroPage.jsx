import { useParams } from "react-router-dom"
import { getHeroeById } from "../helpers";
import { HeroCard } from "../components";

export const HeroPage = () => {

  const { id } = useParams();
  const hero = getHeroeById(id);
  console.log(hero);
  return (
    <>
      <h1>Hero PAge</h1>
      <hr />
      <HeroCard key={ hero.id}/>
    </>
  )
}
