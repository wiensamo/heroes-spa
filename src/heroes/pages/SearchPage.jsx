import { useLocation, useNavigate } from "react-router-dom"
import { useForm } from "../../hooks/useForm"
import { HeroCard } from "../components/HeroCard"
import queryString from 'query-string';
import { getHereosByName } from "../helpers";

export const SearchPage = () => {

const navigate = useNavigate();
const location = useLocation();
const { q='' } = queryString.parse(location.search);
const heroes = getHereosByName(q);

const showSearch = (q.length ===0);
const showError = (q.length>0) && heroes.length===0;

  const { searchText, onInputChange } = useForm({
    searchText: q,
  })
  
  const onSearchSubmit =(event)=>{
    event.preventDefault();
    if( searchText.trim().length <1 ) return;
    navigate(`?q=${searchText}`);
  }
  
  
  return (
    <>
    <h1>Search</h1>
      <hr />
        <div className="row">
          <div className="col-5">
            <h4>Searching</h4>
            <hr />
            <form onSubmit={onSearchSubmit}>
              <input 
                type="text"
                placeholder="Search a Hero"
                className="form-control"
                name="searchText"
                autoComplete="off"
                value={searchText}
                onChange={onInputChange}
              />
              <button className="btn btn-outline-primary mt-2">
                Search
              </button>
            </form>
          </div>

          <div className="col-7">
            <h5>Result</h5>
            <hr /> 
            {/* {
              (q==='')
              ? <div className="alert alert-primary"></div>
              : (heroes.length === 0) &&  <div className="alert alert-danger">No hero with <b>{q}</b> </div>
            } */}
            
            <div className="alert alert-primary animate__animated animate__fadeIn" 
                 style={{display: showSearch ?'':'none'}}>
              Search a hero
            </div>
            <div className="alert alert-danger animate__animated animate__fadeIn" 
                 style={{display: showError ? '':'none'}}>
              No hero with <b>{q}</b> 
            </div>

           
            {
              heroes.map( hero => (
                <HeroCard key={hero.id} {...hero} />
              ))
            }
            {/* {<HeroCard />} */}
          </div>
        </div>
            
    </>
  )
}
