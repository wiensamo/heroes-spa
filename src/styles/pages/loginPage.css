.prueba{
background-color: blue;
color: aquamarine;
}