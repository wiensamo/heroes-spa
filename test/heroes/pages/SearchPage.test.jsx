import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes/pages/SearchPage';
import { AuthContext } from '../../../src/auth/context/AuthContext';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', ()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: ()=> mockedUseNavigate
}));

describe('Test in <SearchPage />', () => { 
    
    beforeEach(()=> jest.clearAllMocks);
    
    test('musht show defaults values correctly', () => { 
        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        );
        expect( container ).toMatchSnapshot();
     });

     test('must be show Batman and the input with queryString value', () => { 
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        );
        const inputValue = screen.getByRole('textbox');
        expect(inputValue.value).toBe('batman');
        const img = screen.getByRole('img')
        expect(img.src).toContain('/assets/heroes/dc-batman.jpg');   
        const alert = screen.getByLabelText('alert-display');
        expect(alert.style.display).toBe('none');
      })

test('must be  show error if dont find hero (batman123)', () => {  
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']} >
                <SearchPage />
             </MemoryRouter>
        );
        const alert = screen.getByLabelText('alert-display');
        expect(alert.style.display).toBe('none');
});


test('must be  called the new page', () => {
    const inputValue = 'superman'
    render(
        <MemoryRouter initialEntries={['/search']} >
            <SearchPage />
         </MemoryRouter>
    );
    const input = screen.getByRole('textbox');
    fireEvent.change(input, {target: {name: 'searchText', value: inputValue}});
    const form = screen.getByRole('form');
    fireEvent.submit(form);
    expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);
    

})

 })