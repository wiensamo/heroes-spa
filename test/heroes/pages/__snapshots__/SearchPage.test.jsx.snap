// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchPage /> musht show defaults values correctly 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h5>
        Result
      </h5>
      <hr />
      <div
        aria-label="alert-display"
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
